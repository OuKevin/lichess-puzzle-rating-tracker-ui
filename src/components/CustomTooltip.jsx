import React from 'react';
import formatDate from '../utils/formatDate';
import get from 'lodash/get';
import './CustomToooltip.css';

const CustomTooltip = ({ active, payload, label }) => {
  if (active) {
    return (
      <div className="custom-tooltip">
        <p>{`Rating : ${get(payload, '0.value', 0)}`}</p>
        <p>{`Date : ${formatDate(label)}`}</p>
      </div>
    );
  }

  return null;
};

export default CustomTooltip;
