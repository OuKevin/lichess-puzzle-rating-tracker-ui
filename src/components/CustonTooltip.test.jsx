import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CustomTooltip from './CustomTooltip'

describe('CustomTooltip', () => {
  const props = {
    active: true,
    payload: [{ value: 100 }],
    label: new Date('2019-03-31T04:59:49.666Z'),
  }
  it('renders active tooltips as expected', () => {
    const wrapper = shallow(<CustomTooltip {...props} />)

    expect(toJson(wrapper)).toMatchSnapshot();
  })
  it('renders inactive tooltips as expected', () => {
    const wrapper = shallow(<CustomTooltip {...props} active={false} />)

    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
