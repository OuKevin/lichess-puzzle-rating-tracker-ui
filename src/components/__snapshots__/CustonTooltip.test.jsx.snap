// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTooltip renders active tooltips as expected 1`] = `
<div
  className="custom-tooltip"
>
  <p>
    Rating : 100
  </p>
  <p>
    Date : 03/30/2019
  </p>
</div>
`;

exports[`CustomTooltip renders inactive tooltips as expected 1`] = `""`;
